Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1260,6 +1263,65 @@ private static void OnIsValidPropertyChanged(DependencyObject d, DependencyPrope
             }
         }
 
+        // Field used for accessing the ItemsSource though the interface ISupportIncrementalLoading
+        private ISupportIncrementalLoading _incrementalItemsSource;
+
+        private bool _canLoadIncrementalData;
+
+        private bool _itemsSourceIsIncremental;
+
+        // Method that loads additional data once the user has scrolled to a certain threashold
+        private async void LoadMoreDataFromIncrementalItemsSource()
+        {
+            if (_itemsSourceIsIncremental && _incrementalItemsSource.HasMoreItems && _canLoadIncrementalData)
+            {
+                DataGridAutomationPeer peer = DataGridAutomationPeer.FromElement(this) as DataGridAutomationPeer;
+                if (peer != null)
+                {
+                    IScrollProvider isp = (IScrollProvider)peer;
+                    if (isp.VerticalScrollPercent >= IncrementalThreshold)
+                    {
+                        _canLoadIncrementalData = false;
+
+                        await _incrementalItemsSource.LoadMoreItemsAsync(0);
+
+                        _canLoadIncrementalData = true;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Identifies the IncrementalThreshold dependency property.
+        /// </summary>
+        public static readonly DependencyProperty IncrementalThresholdProperty = DependencyProperty.Register(
```

Comment:
Why does this need to be adjustable? Why is this based on percentage? The gap will grow as more items as loading. Meaning that 75% for 50 items is quite a bit different from 75% of 500 items. I wouldn't want it reloading when I've reached item 375 of the 500 items. 
I don't see a reason to allow for this to change and it shouldn't rely on percentage. It should be based on how many pixels it has left. So that you're always getting more items when you reach, or are at the bottom

Reply:
I don't really have a personal preference here. The reason why I am using percentage is in order for the new data to always be loaded at the same scroll bar position, which felt natural to me. Other grids I have used (like Telerik's RadDataGrid if I remember correctly now) work this way. I would be happy to check their source code to confirm it. 

The implementation of ISupportIncrementalLoading that I am using fetches more data by paging. Once you hit the threshold it will only fetch by a fixed amount.

The reason I made it adjustable was to provide user flexibility to decide when to load more data. A lower limit of 50% might be a bit ridiculous, but in case someone wants to load new data at the very bottom it provides the option.

Reply:
Doing it the way you suggest, what would you use as a trigger for loading more data? Comparing the number of visible rows to the number that are yet to be displayed is one option could be one option I guess... Any thoughts?

Reply:
Either number of items until the bottom or pixel based. Using percentage changes load location as you get more items